src/System.Private.CoreLib/ILLinkTrim.xml

```diff
@@ -37,5 +37,7 @@
     <type fullname="System.Diagnostics.Tracing.EventPipe*" />
     <!-- Accessed via private reflection and by native code. -->
     <type fullname="System.Diagnostics.Tracing.RuntimeEventSource" />
+    <!-- Accessed via reflection in TraceLogging-style EventSource events. -->
+    <type fullname="*f__AnonymousType*" />
```

Comment:
Presumably the use of anonymous types in coreclr is really low, and of those that are used, they're most likely not getting trimmed out because they're actually used.  But just to be sure, have you validated whether this is keeping anything else around that would otherwise be dropped?

Reply:
Yes, the only material IL diff is the property and get method associated with this type.